code/modules/assembly/assembly_frame.dm

```diff
@@ -165,6 +179,23 @@
 
 		AS.pulsed()
 
+	if(href_list["reorder"])
+		if(!istype(AS))
```

Comment:
Since you're doing this for each of the hrefs, may be worth moving to the locate and moving the lock href above the locate. Remove duplicate code.